packages/fileeditor-extension/src/commands.ts

```diff
@@ -567,7 +567,15 @@ export namespace Commands {
         if (name && widget) {
           const spec = languages.findByName(name);
           if (spec) {
-            widget.content.model.mimeType = spec.mime as string;
+            if (Array.isArray(spec.mime)) {
+              if (spec.mime.length == 0) {
+                widget.content.model.mimeType = 'text/plain';
+              } else {
+                widget.content.model.mimeType = spec.mime[0] as string;
+              }
```

Comment:
We can be more condense for the fallback. And there is a variable to avoid hard coding `text/plain`.

```suggestion
              widget.content.model.mimeType = spec.mime[0] as string ?? IEditorMimeTypeService.defaultMimeType;
```

Reply:
Much nicer indeed!